pyk/src/pyk/tests/test_kastManip.py

```diff
@@ -103,33 +104,39 @@ def test_minimize_term(self):
                 self.assertEqual(actual, expected)
 
 
-class MlPredToBoolTest(TestCase):
+class BoolMlPredConversionsTest(TestCase):
+
+    # TODO: We'd like for bool_to_ml_pred and ml_pred_to_bool to be somewhat invertible.
+    test_data_1 = (
+        ('equals-true', False, KApply(KLabel('#Equals', [Sorts.BOOL, Sorts.GENERATED_TOP_CELL]), [Bool.true, f(a)]), f(a)),
+        ('top-sort-bool', False, KApply(KLabel('#Top', [Sorts.BOOL])), Bool.true),
+        ('top-no-sort', False, KApply('#Top'), Bool.true),
+        ('top-no-sort', False, mlTop(), Bool.true),
+        ('equals-variabl', False, KApply(KLabel('#Equals'), [x, f(a)]), KApply('_==K_', [x, f(a)])),
+        ('equals-true-no-sort', False, KApply(KLabel('#Equals'), [Bool.true, f(a)]), f(a)),
+        ('equals-token', False, KApply(KLabel('#Equals', [KSort('Int'), Sorts.GENERATED_TOP_CELL]), [intToken(3), f(a)]), KApply('_==K_', [intToken(3), f(a)])),
+        ('not-top', False, KApply(KLabel('#Not', [Sorts.GENERATED_TOP_CELL]), [mlTop()]), Bool.notBool(Bool.true)),
+        ('equals-term', True, KApply(KLabel('#Equals'), [f(a), f(x)]), KApply('_==K_', [f(a), f(x)])),
+        ('simplify-and-true', False, KApply(KLabel('#And', [Sorts.GENERATED_TOP_CELL]), [mlEqualsTrue(Bool.true), mlEqualsTrue(Bool.true)]), Bool.true),
+        ('ceil-set-concat-no-sort', True, KApply(KLabel('#Ceil', [KSort('Set'), Sorts.GENERATED_TOP_CELL]), [KApply(KLabel('_Set_'), [KVariable('_'), KVariable('_')])]), KVariable('Ceil_0f9c9997')),
+        ('ceil-set-concat-sort', True, KApply(KLabel('#Not', [Sorts.GENERATED_TOP_CELL]), [KApply(KLabel('#Ceil', [KSort('Set'), Sorts.GENERATED_TOP_CELL]), [KApply(KLabel('_Set_'), [KVariable('_'), KVariable('_')])])]), Bool.notBool(KVariable('Ceil_0f9c9997'))),
+        ('exists-equal-int', True, KApply(KLabel('#Exists', [Sorts.INT, Sorts.BOOL]), [KVariable('X'), KApply('_==Int_', [KVariable('X'), KVariable('Y')])]), KVariable('Exists_6acf2557')),
+    )
+    test_data_2 = (
```

Comment:
Rename, add types, and move to just before the test that uses it.

Reply:
I organized them, but still cannot get the typing to work.

Reply:
Recorded: https://github.com/runtimeverification/k/issues/2801